lib/dialog-renderer/index.tsx

```diff
@@ -1,64 +1,96 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
 import Modal from 'react-modal';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { get } from 'lodash';
 
+import DialogTypes from '../../shared/dialog-types';
 import * as Dialogs from '../dialogs';
+import { closeDialog } from '../state/ui/actions';
+
+import * as S from '../state';
+import * as T from '../types';
+
+type StateProps = {
+  dialogs: T.DialogType[];
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+};
+
+type Props = StateProps & DispatchProps;
+
+export class DialogRenderer extends Component<Props> {
+  static displayName = 'DialogRenderer';
+
+  static propTypes = {
+    appProps: PropTypes.object.isRequired,
+    buckets: PropTypes.object,
+    themeClass: PropTypes.string,
+    isElectron: PropTypes.bool.isRequired,
+    isMacApp: PropTypes.bool.isRequired,
+  };
+
+  renderDialog(dialogType: string) {
+    const {
+      appProps,
+      buckets,
+      themeClass,
+      isElectron,
+      isMacApp,
+      closeDialog,
+    } = this.props;
+
+    const dialog = get(DialogTypes, dialogType);
+
+    if (dialog === null) {
+      throw new Error('Unknown dialog type.');
+    }
 
-export const DialogRenderer = props => {
-  const {
-    appProps,
-    buckets,
-    themeClass,
-    closeDialog,
-    dialogs,
-    isElectron,
-    isMacApp,
-  } = props;
-
-  const renderDialog = dialog => {
-    const { key, title } = dialog;
     const DialogComponent = Dialogs[dialog.type];
```

Comment:
This little bit of magic shouldn't work and probably only does because of the fact that `webpack` isn't entirely doing what it's supposed to be doing. We can fix the imported `../dialogs` file but probably the best thing to do would be to `case` this for now, which would also remove that additional layer of implicit logic or string-based magic.

```ts
import * as Dialogs from '../dialogs'

const getDialogComponent = type => {
  switch (type) {
    case 'About': return Dialogs.About;
    case 'Import': return Dialogs.Import;
    case 'Settings': return Dialogs.Settings;
    case 'Share': return Dialogs.Share;
  }
}
```

we could even make the function look like the variable itself…

```ts
const DialogComponent = type => {
  switch (type) {
    case 'About': return Dialogs.About;
    case 'Import': return Dialogs.Import;
    case 'Settings': return Dialogs.Settings;
    case 'Share': return Dialogs.Share;
  }
}

<DialogComponent(dialogType) buckets=… />
```

Reply:
Update: less magic 😿